shared/editor/rules/mention.ts

```diff
@@ -0,0 +1,82 @@
+import MarkdownIt from "markdown-it";
+import StateCore from "markdown-it/lib/rules_core/state_core";
+import Token from "markdown-it/lib/token";
+
+function renderMention(tokens: Token[], idx: number) {
+  const mType = tokens[idx].attrGet("data-type");
+  const mId = tokens[idx].attrGet("data-id");
+  const mActor = tokens[idx].attrGet("data-actor");
+  const label = tokens[idx].content;
+  const id = tokens[idx].attrGet("id");
+
+  return `<span class="mention" id="${id}" data-actor="${mActor}" data-type="${mType}" data-id="${mId}">${label}</span>`;
+}
+
+function parseMentions(state: StateCore) {
```

Comment:
I think you should revert the last changes to store the actor Id in the Markdown, it shouldn't be needed at all and complicates the format a lot